File: NumberOf1Bits/Solution.js
/**
 * @param {number} n - a positive integer
 * @return {number}
 */
var hammingWeight = function(n) {
    var mask = 0x0001;
    var num = 0;
    for (var i = 0; i < 32; i++) {
        if ((n & mask) !== 0) num++;
        mask = mask << 1;
    }
    return num;
};

File: NumberOf1Bits/Solution.rb
# @param {Integer} n, a positive integer
# @return {Integer}
def hamming_weight(n)
    mask = 0x0001
    num = 0
    i = 0
    while i < 32
        num += 1 if (n & mask) != 0
        mask = mask << 1
        i += 1
    end
    return num
end
